indico/modules/attachments/controllers/display/event.py

```diff
@@ -53,3 +55,16 @@ def wp(self):
             return WPPackageEventAttachmentsDisplayConference
         else:
             return WPPackageEventAttachmentsDisplay
+
+
+class RHPackageEventAttachmentsStatus(RHDisplayEventBase):
+    def _process(self):
+        res = AsyncResult(request.view_args['task_id'])
+        try:
+            download_url = res.get(5, propagate=False)
+        except TimeoutError:
+            return jsonify(pending=True)
+        if res.successful():
+            return {'pending': False, 'download_url': download_url}
+        else:
+            raise RuntimeError('Material package generation failed')
```

Comment:
We could discard the result once we are done since we'll never need it again anyway:

```suggestion
        try:
            if res.successful():
                return {'pending': False, 'download_url': download_url}
            else:
                raise RuntimeError('Material package generation failed')
        finally:
            res.forget()
```

Reply:
`IndicoError` instead of `RuntimeError` looks nicer for the user (no `RuntimeError:` prefix in the message)